src/configs/config.js

```diff
@@ -0,0 +1,30 @@
+// can add more configuration for prod environment
+const env = process.env.NODE_ENV; // 'dev' or 'test'
+
+const dev = {
+    app: {
+        port: parseInt(process.env.DEV_APP_PORT) || 3000
+    },
+    db: {
+        host: process.env.DEV_DB_HOST || 'localhost',
+        port: parseInt(process.env.DEV_DB_PORT) || 27017,
+        name: process.env.DEV_DB_NAME || 'dev'
+    }
+};
+const test = {
+    app: {
+        port: parseInt(process.env.TEST_APP_PORT) || 3000
+    },
+    db: {
+        host: process.env.TEST_DB_HOST || 'localhost',
+        port: parseInt(process.env.TEST_DB_PORT) || 27017,
+        name: process.env.TEST_DB_NAME || 'test'
+    }
+};
+
+const config = {
+    dev,
+    test
+};
```

Comment:
when to use `dev` config, and when to use `test` config?
Image that this app will be deployed to staging environment, how to configure the app with environment variables?
And in staging environment, the database requires username / password authentication, how to configure the app?

Reply:
When you need some test data for local development, use `dev` config; when you need to run integration tests to check if the data actually hits the database, use `test` config.  

For staging environment, I would add a file which contains encrypted credentials to the machine hosting the staging environment, and add `staging` config to the app which reads username & password from the file.

Reply:
this question is programming language independent, and database independent, 
the connection string in `app.js` doesn't read username / password, 
https://github.com/Wiredcraft/test-backend/blob/848b1d715d1673c2a4072ea01a4d4d842e9b7a15/src/app.js#L8-L9
I think simply replace `config.js` file won't work.

Reply:
I see. Regardless of programming language and database, I think the best practice is to put sensitive data in the file such as `db.yaml` on the host machine, and when app starts, it reads everything from the file based on different environments.  If using environment variables, for this app, it should be something like:
```
const { db: { username, password, host, port, name } } = config; 
const connectionString = `mongodb://${username}:${password}${host}:${port}/${name}`; 
```